Objects/setobject.c

```diff
@@ -535,9 +535,16 @@ set_repr_lock_held(PySetObject *so)
         return PyUnicode_FromFormat("%s()", Py_TYPE(so)->tp_name);
     }
 
-    keys = PySequence_List((PyObject *)so);
-    if (keys == NULL)
+    keys = PyList_New(so->used);
```

Comment:
I think this probably deserves a comment even though it's probably faster anyway.

```suggestion
    // Avoid PySequence_List because it might re-lock the object lock or
    // the GIL and allow something to clear the set from underneath us.
    keys = PyList_New(so->used);
```